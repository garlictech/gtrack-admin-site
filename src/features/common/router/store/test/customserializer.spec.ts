import { CustomSerializer } from '../';

describe('Test the custom serializer', () => {
  const serializer = new CustomSerializer();

  const routerState: any = {
    url: 'fooUrl',
    root: {
      queryParams: { foo: 'querybar' },
      url: 'rootUrl',
      params: { foo: 'paramsbar' },
      data: { foo: 'databar' }
    }
  };

  it('would serialize the router state', () => {
    expect(serializer.serialize(routerState)).toMatchSnapshot();
  });

  it('would serialize the router state with firstChild', () => {
    const _routerState = { ...routerState };
    _routerState.root.firstChild = { ...routerState.root, params: { foo: 'otherparam' } };
    expect(serializer.serialize(_routerState)).toMatchSnapshot();
  });
});
