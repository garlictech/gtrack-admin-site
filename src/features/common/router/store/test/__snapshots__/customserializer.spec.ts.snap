// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the custom serializer would serialize the router state 1`] = `
Object {
  "data": Object {
    "foo": "databar",
  },
  "params": Object {
    "foo": "paramsbar",
  },
  "queryParams": Object {
    "foo": "querybar",
  },
  "url": "fooUrl",
}
`;

exports[`Test the custom serializer would serialize the router state with firstChild 1`] = `
Object {
  "data": Object {
    "foo": "databar",
  },
  "params": Object {
    "foo": "otherparam",
  },
  "queryParams": Object {
    "foo": "querybar",
  },
  "url": "fooUrl",
}
`;
